src/utils/http/basicHttpClient.ts

```diff
@@ -79,6 +79,10 @@ export class BasicHttpClient {
     return this;
   }
 
+  public getDefaultHeaders(): HttpHeaders {
+    return this.defaultHeaders;
```

Comment:
need to clone here as well?
```suggestion
    return { ...this.defaultHeaders };
```